core/math/math_funcs.h

```diff
@@ -388,7 +388,7 @@ class Math {
 	}
 
 	static _ALWAYS_INLINE_ bool is_zero_approx(double s) {
-		return abs(s) < CMP_EPSILON;
+		return abs(s) < CMP_EPSILON_64;
```

Comment:
It's likely that somewhere in the engine the value can get truncated to 32-bit floats. Having a stricter epsilon can be useful for debugging, but I wouldn't enable it by default.

```suggestion
#ifdef PRECISE_MATH_CHECKS
		return abs(s) < CMP_EPSILON_64;
#else
		return abs(s) < CMP_EPSILON;
#endif // PRECISE_MATH_CHECKS
```

Reply:
IMO, if the `double` version of `is_zero_approx` is used in a version of Godot compiled with doubles, and yet you want such a high epsilon to be used due to some kind of truncation, it means the code calling it is wrong, or it should be using the `float` version all the time. I don't think hiding the issue solves it.

Reply:
All good points, but I am thinking it would be a risky change. As you mentioned, some code calling it could be wrong, or somewhere down the line some piece of code can be truncating to 32-bit floats. It seems like these kinds of problems are worth finding and fixing, but doing it by default seems risky if a bug appears in production. I think we'll have to ask @akien-mga or @reduz what their thoughts are on this.

Reply:
Maybe we should only use the 64-bit epsilon if we both pass in doubles and Godot is compiled with doubles?

```suggestion
#ifdef REAL_T_IS_DOUBLE
		return abs(s) < CMP_EPSILON_64;
#else
		return abs(s) < CMP_EPSILON;
#endif // REAL_T_IS_DOUBLE
```